### PR TITLE
mbedtls: add mbedtls alternative implementation modules

### PR DESCRIPTION
**Pull request by ThePassionate**
(1)use new mbedtls_config.h
(2)add sha1/sha256/sha512 alternative implementation from /dev/crypto
(3)add hardware entropy from /dev/random
Signed-off-by: makejian <makejian@xiaomi.com>

### CODE REVIEW

File: crypto/mbedtls/Make.defs [hunk 1]
```diff
@@ -23,10 +23,13 @@ CONFIGURED_APPS += $(APPDIR)/crypto/mbedtls
 
 # Allows `<mbedtls/<>.h>` import.
 
-CFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/mbedtls/include
-CXXFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/mbedtls/include
+MBEDTLS_FLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/include
+MBEDTLS_FLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/mbedtls/include
 
-CFLAGS += ${DEFINE_PREFIX}MBEDTLS_CONFIG_FILE="<crypto/mbedtls_config.h>"
```
[the remark this answers is not in the document]
Comment (ThePassionate) [reply crypto/mbedtls/Make.defs:29]: done

**Review by xiaoxiang781216 — APPROVED** [review]

File: crypto/mbedtls/source/entropy_alt.c [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#include <stdio.h>
+#include <unistd.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int mbedtls_hardware_poll(FAR void *data,
+                          FAR unsigned char *output,
+                          size_t len, size_t *olen)
```
Comment (pkarashchenko) [inline crypto/mbedtls/source/entropy_alt.c:35]: `FAR size_t *olen)`

File: testing/crypto/hash.c [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+{
+  FAR char *data;
+  int datalen;
+}tb;
+
+tb md5_testcase[] =
+{
+    {
+      "",
+      0,
```
Comment (pkarashchenko) [inline testing/crypto/hash.c:52]: remove 2 leading spaces

File: testing/crypto/hash.c [hunk 4]
```diff
@@ -0,0 +1,535 @@
...
+      0x81, 0xa1, 0xc7, 0xe2, 0x84, 0xd7, 0x3e, 0x67,
+      0xf1, 0x80, 0x9a, 0x48, 0xa4, 0x97, 0x20, 0x0e,
+      0x04, 0x6d, 0x39, 0xcc, 0xc7, 0x11, 0x2c, 0xd0 }
+};
+
+/* FIPS-180-2 test vectors */
+
+static const unsigned char sha512_result[3][64] =
+{
+    { 0xdd, 0xaf, 0x35, 0xa1, 0x93, 0x61, 0x7a, 0xba,
```
Comment (pkarashchenko) [inline testing/crypto/hash.c:168]: remove 2 leading spaces

File: testing/crypto/hash.c [hunk 5]
```diff
@@ -0,0 +1,535 @@
...
+
+  return 0;
+}
+
+static int match(const unsigned char *a, const unsigned char *b, size_t len)
+{
+  int i;
+
+  if (memcmp(a, b, len) == 0)
+    return (0);
```
Comment (pkarashchenko) [inline testing/crypto/hash.c:284]:
```suggestion
  if (memcmp(a, b, len) == 0)
    {
       return 0;
     }
```

**Pull request merged**